0e9327b9f23233f939495ef96346abf3844d73a4..9673a36727ea6ccf4676f0db31d0c468aba33e94

From da42cf905d07613ad5243ddee16486369025eb6e Mon Sep 17 00:00:00 2001
From: Henri Verbeet <hverbeet@codeweavers.com>
Date: Wed, 8 May 2019 17:51:36 +0430
Subject: [PATCH] wined3d: Store the glEnableWINE and glDisableWINE pointers in
 the wined3d_gl_info structure.

Signed-off-by: Henri Verbeet <hverbeet@codeweavers.com>
Signed-off-by: Alexandre Julliard <julliard@winehq.org>
---

 dlls/wined3d/context.c         | 2 +-
 dlls/wined3d/directx.c         | 4 ++--
 dlls/wined3d/gl_compat.c       | 8 ++++----
 dlls/wined3d/state.c           | 4 ++--
 dlls/wined3d/wined3d_gl.h      | 2 --
 dlls/wined3d/wined3d_private.h | 3 +++
 6 files changed, 12 insertions(+), 11 deletions(-)


diff --git a/dlls/wined3d/context.c b/dlls/wined3d/context.c
index 664ef732b40..b9a2f4e9b00 100644
--- a/dlls/wined3d/context.c
+++ b/dlls/wined3d/context.c
@@ -2053,5 +2053,5 @@ static void SetupForBlit(const struct wined3d_device *device, struct wined3d_con
     checkGLcall("glDisable GL_DEPTH_TEST");
     context_invalidate_state(context, STATE_RENDER(WINED3D_RS_ZENABLE));
-    glDisableWINE(GL_FOG);
+    gl_info->p_glDisableWINE(GL_FOG);
     checkGLcall("glDisable GL_FOG");
     context_invalidate_state(context, STATE_RENDER(WINED3D_RS_FOGENABLE));
diff --git a/dlls/wined3d/directx.c b/dlls/wined3d/directx.c
index 5f995ced018..e33f2499734 100644
--- a/dlls/wined3d/directx.c
+++ b/dlls/wined3d/directx.c
@@ -5872,6 +5872,6 @@ static BOOL wined3d_adapter_init(struct wined3d_adapter *adapter, UINT ordinal,
 #endif
 
-    glEnableWINE = gl_info->gl_ops.gl.p_glEnable;
-    glDisableWINE = gl_info->gl_ops.gl.p_glDisable;
+    gl_info->p_glEnableWINE = gl_info->gl_ops.gl.p_glEnable;
+    gl_info->p_glDisableWINE = gl_info->gl_ops.gl.p_glDisable;
 
     if (!AllocateLocallyUniqueId(&adapter->luid))
diff --git a/dlls/wined3d/gl_compat.c b/dlls/wined3d/gl_compat.c
index 7e6b5101bc4..a2fb09c39e1 100644
--- a/dlls/wined3d/gl_compat.c
+++ b/dlls/wined3d/gl_compat.c
@@ -438,8 +438,8 @@ void install_gl_compat_wrapper(struct wined3d_gl_info *gl_info, enum wined3d_gl_
             old_fogcoord_glFogfv = gl_info->gl_ops.gl.p_glFogfv;
             gl_info->gl_ops.gl.p_glFogfv = wine_glFogfv;
-            old_fogcoord_glEnable = glEnableWINE;
-            glEnableWINE = wine_glEnable;
-            old_fogcoord_glDisable = glDisableWINE;
-            glDisableWINE = wine_glDisable;
+            old_fogcoord_glEnable = gl_info->p_glEnableWINE;
+            gl_info->p_glEnableWINE = wine_glEnable;
+            old_fogcoord_glDisable = gl_info->p_glDisableWINE;
+            gl_info->p_glDisableWINE = wine_glDisable;
 
             old_fogcoord_glVertex4f = gl_info->gl_ops.gl.p_glVertex4f;
diff --git a/dlls/wined3d/state.c b/dlls/wined3d/state.c
index 903f0d4e4b7..ec5e89d0d98 100644
--- a/dlls/wined3d/state.c
+++ b/dlls/wined3d/state.c
@@ -1031,5 +1031,5 @@ void state_fog_fragpart(struct wined3d_context *context, const struct wined3d_st
     {
         /* No fog? Disable it, and we're done :-) */
-        glDisableWINE(GL_FOG);
+        gl_info->p_glDisableWINE(GL_FOG);
         checkGLcall("glDisable GL_FOG");
         return;
@@ -1168,5 +1168,5 @@ void state_fog_fragpart(struct wined3d_context *context, const struct wined3d_st
     }
 
-    glEnableWINE(GL_FOG);
+    gl_info->p_glEnableWINE(GL_FOG);
     checkGLcall("glEnable GL_FOG");
     if (new_source != context->fog_source || fogstart == fogend)
diff --git a/dlls/wined3d/wined3d_gl.h b/dlls/wined3d/wined3d_gl.h
index be4e30851cb..3bc0d7e265d 100644
--- a/dlls/wined3d/wined3d_gl.h
+++ b/dlls/wined3d/wined3d_gl.h
@@ -30,6 +30,4 @@
 #define GL_COMPRESSED_LUMINANCE_ALPHA_3DC_ATI 0x8837  /* not in the gl spec */
 
-void (WINE_GLAPI *glDisableWINE)(GLenum cap) DECLSPEC_HIDDEN;
-void (WINE_GLAPI *glEnableWINE)(GLenum cap) DECLSPEC_HIDDEN;
 
 /* OpenGL extensions. */
diff --git a/dlls/wined3d/wined3d_private.h b/dlls/wined3d/wined3d_private.h
index 81c3af20a39..790d7221b18 100644
--- a/dlls/wined3d/wined3d_private.h
+++ b/dlls/wined3d/wined3d_private.h
@@ -1830,4 +1830,7 @@ struct wined3d_gl_info
     struct wined3d_format *formats;
     unsigned int format_count;
+
+    void (WINE_GLAPI *p_glDisableWINE)(GLenum cap);
+    void (WINE_GLAPI *p_glEnableWINE)(GLenum cap);
 };
 
--
